apps/prairielearn/src/lib/cache.ts

```diff
@@ -1,138 +1,132 @@
     cache = new LRUCache({
-      max: config.questionRenderCacheMaxItems,
+      // The in-memory cache is really only suited for development, so we'll
+      // hardcode a relatively low limit here.
+      max: 1000,
     });
   } else {
-    cacheEnabled = false;
-    return callback(new Error(`Unknown cache type "${cacheType}"`));
+    throw new Error(`Unknown cache type "${cacheType}"`);
   }
-
-  callback(null);
 }
 
-export function set(key: string, value: any, maxAgeMS = null) {
+export function set(key: string, value: string | number | boolean, maxAgeMS?: number) {
   if (!cacheEnabled) return;
 
   switch (cacheType) {
-    case 'memory':
-      if (maxAgeMS) {
-        cache.set(key, value, { ttl: maxAgeMS });
-      } else {
-        cache.set(key, value);
-      }
+    case 'memory': {
+      cache.set(key, JSON.stringify(value), { ttl: maxAgeMS ?? undefined });
       break;
-    case 'redis':
+    }
+
+    case 'redis': {
       // This returns a promise, but we don't want to wait for this data
       // to reach the cache before continuing, and we don't *really*
-      // care if it errors
-      if (maxAgeMS) {
-        // we don't log the error because it contains the cached value, which can be huge and which fills up the logs
-        cache
-          .set(key, value, maxAgeMS)
-          .catch((_err) => logger.error('Cache set error', { key, maxAgeMS }));
-      } else {
-        // we don't log the error because it contains the cached value, which can be huge and which fills up the logs
-        cache.set(key, value).catch((_err) => logger.error('Cache set error', { key }));
-      }
+      // care if it errors.
+      //
+      // We don't log the error because it contains the cached value,
+      // which can be huge and which fills up the logs.
+      client
+        .set(key, JSON.stringify(value), { PX: maxAgeMS ?? undefined })
+        .catch((_err) => logger.error('Cache set error', { key, maxAgeMS }));
       break;
-    default:
-      throw new Error(`Unknown cache type "${cacheType}"`);
+    }
   }
 }
 
 /**
- * Calls the callback with the value if it exists in the cache, or null otherwise.
+ * Returns the value for the corresponding key if it exists in the cache; null otherwise.
  */
-export async function getAsync(key: string) {
+export async function get(key: string): Promise<any> {
   if (!cacheEnabled) return null;
 
   switch (cacheType) {
-    case 'memory':
-      return cache.get(key) ?? null;
-    case 'redis':
-      return cache.get(key);
-    default:
-      throw new Error(`Unknown cache type "${cacheType}"`);
+    case 'memory': {
+      const value = cache.get(key);
+      if (typeof value === 'string') {
+        return JSON.parse(value);
+      }
+      return null;
+    }
+
+    case 'redis': {
+      const value = await client.get(key);
+      if (typeof value === 'string') {
+        return JSON.parse(value);
+      }
+      return null;
+    }
+
+    default: {
+      return null;
```

Comment:
It's a super-minor thing, but I think the convention for built-in JS data types like `Map` is to [return `undefined`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get#return_value) if the key is not present.

Reply:
Nice! I'll go with that.